spanner/read_test.go

```diff
@@ -1672,6 +1672,75 @@ func setupStatementResult(t *testing.T, server *MockedSpannerInMemTestServer, st
 	return server.TestSpanner.PutStatementResult(stmt, result)
 }
 
+func TestSkippingTrailersForExecuteStreamingSql(t *testing.T) {
+	t.Parallel()
+
+	server, client, teardown := setupMockedTestServer(t)
+	res := server.CreateSingersResults(4, true)
+	sql := "SELECT SingerId, AlbumId, AlbumTitle FROM Albums WHERE 1=2"
+	err := server.TestSpanner.PutStatementResult(sql, res)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardown()
+	ctx := context.Background()
+	iter := client.Single().Query(ctx, NewStatement(sql))
+	defer iter.Stop()
+	var noOfRows int
+	for {
+		row, err := iter.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		noOfRows++
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if row.ColumnValue(0).GetStringValue() == "0" {
+			t.Fatalf("ID Mismatch in the result")
+		}
+	}
+	if noOfRows != 4 {
+		t.Fatalf("no of rows mismatch, Expected %v Got %v", 1, noOfRows)
+	}
+}
+
+func TestSkippingTrailersForStreamingRead(t *testing.T) {
+	t.Parallel()
+
+	server, client, teardown := setupMockedTestServer(t)
+
+	res := server.CreateSingersResults(5, true)
+	sql := "SELECT SingerId, AlbumId, AlbumTitle FROM Albums"
+	err := server.TestSpanner.PutStatementResult(sql, res)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardown()
+	ctx := context.Background()
+	iter := client.Single().Read(ctx, "Albums", KeySets(Key{"foo"}), []string{"SingerId", "AlbumId", "AlbumTitle"})
+	defer iter.Stop()
+	var noOfRows int
+	for {
+		row, err := iter.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		noOfRows++
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if row.ColumnValue(0).GetStringValue() == "0" {
```

Comment:
I think this would be a more relevant check:

```suggestion
		if row.ColumnValue(0).GetStringValue() != fmt.Sprintf("%v", noOfRows) {
```

Reply:
(Same for the other test function)

Reply:
done